engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventTest.java

```diff
@@ -134,6 +137,69 @@ public void shouldNotDeployCompensationHandlerWithBoundaryEvent() {
             Task.class, "A compensation handler should have no boundary events"));
   }
 
+  @Test
+  public void shouldDeployCompensationEventSubprocess() {
+    final var process =
+        Bpmn.createExecutableProcess("compensation-process")
+            .startEvent()
+            .subProcess(
+                "subprocess",
+                s ->
+                    s.embeddedSubProcess()
+                        .eventSubProcess(
+                            "embedded-subprocess",
+                            eventSubProcess ->
+                                eventSubProcess
+                                    .startEvent(
+                                        "compensation-start",
+                                        AbstractStartEventBuilder::compensation)
+                                    .intermediateThrowEvent(
+                                        "compensation-throw-event",
+                                        AbstractThrowEventBuilder::compensateEventDefinition)
+                                    .userTask("B")
+                                    .endEvent())
+                        .startEvent()
+                        .userTask("A")
+                        .boundaryEvent()
+                        .compensation(c -> c.userTask("undo-A"))
+                        .endEvent()
+                        .done())
+            .endEvent(
+                "compensation-end-event",
+                end -> end.compensateEventDefinition().compensateEventDefinitionDone())
+            .done();
+
+    ProcessValidationUtil.validateProcess(process);
+  }
+
+  @Test
+  public void shouldNotDeployCompensationEventSubprocessOnProcessLevel() {
+    final var process =
+        Bpmn.createExecutableProcess("compensation-process")
+            .startEvent()
+            .subProcess(
+                "subprocess",
+                s ->
+                    s.embeddedSubProcess()
+                        .startEvent("compensation-start", AbstractStartEventBuilder::compensation)
+                        .intermediateThrowEvent(
+                            "compensation-throw-event",
+                            AbstractThrowEventBuilder::compensateEventDefinition)
+                        .userTask("B")
+                        .endEvent()
+                        .done())
+            .endEvent(
+                "compensation-end-event",
+                end -> end.compensateEventDefinition().compensateEventDefinitionDone())
+            .done();
```

Comment:
:x: This model doesn't create a process with a compensation event subprocess. Instead, it creates an embedded subprocess with a compensation start event. 

![Screenshot from 2023-11-22 09-09-59](https://github.com/camunda/zeebe/assets/4305769/c478c845-b66b-4de7-9205-f564c95afea2)

Reply:
How the process should look like ?

Reply:
We want to restrict that a compensation event subprocess is on the process level. Why? Because we don't propagate the compensation for child process instances. As a result, a compensation event subprocess couldn't be triggered. 

To verify the restriction, the test case should use a process that has a compensation event subprocess that is not embedded in a subprocess.

![Screenshot from 2023-11-22 12-49-18](https://github.com/camunda/zeebe/assets/4305769/b328f2cf-ffb2-4b42-a3c8-791dc70f068e)